File: ChessProject-Java/src/main/java/com/solarwindsmsp/chess/Pawn.java
package com.solarwindsmsp.chess;

public class Pawn extends Piece{

    public Pawn(PieceColor pieceColor) {
        super(pieceColor);
    }

    public boolean move(int newX, int newY) {
        if(getChessBoard() == null) {
            return false;
        }

        if(getXCoordinate() == newX && getYCoordinate() - newY == 1) {
            if(getChessBoard().isLegalBoardPosition(newX, newY)) {
                getChessBoard().movePiece(this, newX, newY);
                setYCoordinate(newY);
                return true;
            }
        }
        return false;
    }
}

File: ChessProject-Csharp/src/Pieces/Pawn.cs
using src.Enums;
using System;

namespace src.Pieces
{
    public class Pawn : Piece
    {
        public Pawn(PieceColor pieceColor) : base(pieceColor, PieceType.Pawn)
        {
        }

        public override bool Move(MovementType movementType, int newX, int newY)
        {
            var result = false;
            if (ChessBoard.IsLegalBoardPosition(newX, newY) && ChessBoard.IsSquareEmpty(newX, newY))
            {
                if (movementType == MovementType.Move)
                {
                    if (this.XCoordinate == newX
                        && (this.PieceColor == PieceColor.Black && newY == this.YCoordinate - 1
                            || this.PieceColor == PieceColor.White && newY == this.YCoordinate + 1))
                    {
                        result = ChessBoard.MovePiece(this, newX, newY, movementType);
                        this.YCoordinate = newY;
                    }
                }
                else
                {
                    throw new NotImplementedException("Only Move movement implemented");
                }
            }
            return result;
        }
    }
}
